simple_salesforce/tests/test_api.py

```diff
@@ -859,8 +890,9 @@ def test_md_deploy_failed_status_code(self, mock_read_zip):
         client = Salesforce(session_id=tests.SESSION_ID,
                             instance=tests.INSTANCE_URL,
                             session=session)
+        zip_path = Path(__file__).parent / 'sample.zip'
         with self.assertRaises(Exception):
-            client.deploy("path/to/fake/zip.zip", sandbox=False)
+            client.deploy(str(zip_path), sandbox=False)
```

Comment:
Changes tests above to use an actual minimal zip file, instead of mocking _read_deploy_zip, and added another test passing an opened file handler instead of a path. This adds coverage to the _read_deploy_zip method.